analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+    )
+
+    tmp_sub_date_severecovid19_opa = (
+        opa_diag.where(
+            ((opa_diag.primary_diagnosis_code.is_in(covid_codes)) | (opa_diag.secondary_diagnosis_code_1.is_in(covid_codes))) & 
+            (opa_diag.appointment_date.is_on_or_after(index_date))
+        )
+        .sort_by(opa_diag.appointment_date)
+        .first_for_patient()
+        .appointment_date
+    )
+
+    ## prepare for smoking status
+
+    tmp_most_recent_smoking_cat = (
+        last_matching_event_clinical_ctv3_before(smoking_clear, index_date)
+        .ctv3_code.to_category(smoking_clear)
+        )
+
+    tmp_ever_smoked = ever_matching_event_clinical_ctv3_before(ever_current_smoke, index_date)   # uses a different codelist with ONLY smoking codes
+
+
+    ## Combine the variables into the final dictionary
+    dynamic_variables = dict(
+
+# Exposures---------------------------------------------------------------------------------------------------
+
+        ### Exposures----Covid-19
+
+        tmp_exp_date_covid19_confirmed_sgss=tmp_exp_date_covid19_confirmed_sgss,
+        tmp_exp_date_covid19_confirmed_snomed=tmp_exp_date_covid19_confirmed_snomed,
+        tmp_exp_date_covid19_confirmed_apc=tmp_exp_date_covid19_confirmed_apc,
+        tmp_exp_date_covid19_confirmed_opa=tmp_exp_date_covid19_confirmed_opa,
+        tmp_exp_date_death=tmp_exp_date_death,
+        tmp_exp_covid19_confirmed_death=tmp_exp_covid19_confirmed_death,
+        tmp_exp_date_covid19_confirmed_death=tmp_exp_date_covid19_confirmed_death,
+        
+        exp_date_covid19_confirmed=minimum_of(
+            tmp_exp_date_covid19_confirmed_sgss, 
+            tmp_exp_date_covid19_confirmed_snomed,
+            tmp_exp_date_covid19_confirmed_apc,
+            tmp_exp_date_covid19_confirmed_opa,
+            tmp_exp_date_covid19_confirmed_death
+        ),
+
+
+
+# Outcomes---------------------------------------------------------------------------------------------------
+
+    ## ---First recording of the outcome in during the study period
+
+      ### Pneumonia
+        out_date_pneumonia= (
+            first_matching_event_clinical_snomed_between(
+            pneumonia_snomed, index_date, end_date_out
+            )
+        .date
+        ),
+
+      ### Asthma
+        out_date_asthma= (
+            first_matching_event_clinical_snomed_between(
+            asthma_snomed, index_date, end_date_out
+            )
+        .date
+        ),
+
+      ### COPD
+        out_date_copd= (
+            first_matching_event_clinical_ctv3_between(
+            copd_ctv3, index_date, end_date_out
+            )
+        .date
+        ),
+
+      ### Pulmonary Fibrosis
+        out_date_pulmonary_fibrosis= (
+            first_matching_event_clinical_snomed_between(
+            pulmonary_fibrosis_snomed, index_date, end_date_out
+            )
+        .date
+        ),
+
+# DEFINE EXISTING RESPIRATORY CONDITION COHORT --------------------------------------------------------------
```

Comment:
Existing respiratory conditions may also need to be defined in primary and secondary care, depending on the decision about outcomes.

Reply:
Decision taken 03/12/2024 to include outcomes from primary care, secondary care and deaths as per other protocols. Recorded in issue https://github.com/opensafely/post-covid-respiratory/issues/45 and to be resolved in separate PR.